clients/deltalake/src/main/scala/org/projectnessie/deltalake/NessieLogStore.scala

```diff
@@ -99,8 +100,8 @@ class NessieLogStore(sparkConf: SparkConf, hadoopConf: Configuration)
 
     try {
       val ref = Option(requestedRef)
-        .map(client.getTreeApi.getReferenceByName(_))
-        .getOrElse(client.getTreeApi.getDefaultBranch)
+        .map(api.getReference.refName(_).submit())
```

Comment:
nit: `.submit()` looks odd in this context... I get the impression that the result is received asynchronously. Would `.retrieve()` be more readable?

This is not really related to the removal of the old `NessieClient`, but more of a general API clean-up comment.

Reply:
Mind opening a ticket for that to be done when everything else is merged?

Reply:
+1 on updating `sumit` to `retrieve` or `get`

Reply:
In a different PR IMO. This one is updating the call-sites, not the API itself.

Reply:
Different PR, sure. I'm fine with `get` too.

Reply:
#2029